operator/v1/types_console.go

```diff
@@ -55,15 +55,22 @@ type ConsoleSpec struct {
 	// +optional
 	Route ConsoleConfigRoute `json:"route"`
 	// plugins defines a list of enabled console plugin names.
+	// +listType=atomic
+	// +kubebuilder:validation:MaxItems=64
+	// +kubebuilder:validation:XValidation:rule="self.all(x, self.exists_one(y, x == y))",message="each plugin name must be unique"
 	// +optional
-	Plugins []string `json:"plugins,omitempty"`
+	Plugins []PluginName `json:"plugins,omitempty"`
 	// ingress allows to configure the alternative ingress for the console.
 	// This field is intended for clusters without ingress capability,
 	// where access to routes is not possible.
 	// +optional
 	Ingress Ingress `json:"ingress"`
 }
 
+// +kubebuilder:validation:MaxLength=128
+// +kubebuilder:validation:Pattern=^[a-zA-Z0-9-]+$
+type PluginName string
```

Comment:
This needs godoc with some more information about the type and the validations on it. Good guideline for writing human readable go doc can be found here: https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#write-user-readable-documentation-in-godoc

Reply:
Let's make sure that the detailed documentation ends up on the field itself, rather than the type alias.

The CRD schema will pick up godoc from the field, but sadly not the type alias.